controllers/spiaccesstoken_controller.go

```diff
@@ -329,21 +349,21 @@ func (f *linkedBindingsFinalizer) Finalize(ctx context.Context, obj client.Objec
 	}
 }
 
-func (f *linkedBindingsFinalizer) hasLinkedBindings(ctx context.Context, token *api.SPIAccessToken) (bool, error) {
+func (f *tokenStorageFinalizer) Finalize(ctx context.Context, obj client.Object) (finalizer.Result, error) {
+	err := f.storage.Delete(ctx, obj.(*api.SPIAccessToken))
+	if err != nil {
+		err = fmt.Errorf("failed to delete the linked token during finalization of %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
+	}
+	return finalizer.Result{}, err
+}
+
+func hasLinkedBindings(ctx context.Context, token *api.SPIAccessToken, client_ client.Client) (bool, error) {
 	list := &api.SPIAccessTokenBindingList{}
-	if err := f.client.List(ctx, list, client.InNamespace(token.Namespace), client.Limit(1), client.MatchingLabels{
+	if err := client_.List(ctx, list, client.InNamespace(token.Namespace), client.Limit(1), client.MatchingLabels{
```

Comment:
`client_` looks weird. Maybe name it something like `k8sClient` or rename linked package to something like `k8s "sigs.k8s.io/controller-runtime/pkg/client"` ?

Reply:
ok